devito/symbolics/manipulation.py

```diff
@@ -94,6 +107,53 @@ def _eval_numbers(expr, args):
         args[:] = [expr.func(*numbers)] + others
 
 
+class Uxmapper(dict):
+
+    """
+    A helper mapper for uxreplace. Any dict can be passed as subs to uxreplace,
+    but with a Uxmapper it is also possible to easily express compound replacements,
+    such as sub-expressions within n-ary operations (e.g., `a*b` inside the 4-way
+    Mul `a*c*b*d`).
+    """
+
+    class Uxsubmap(dict):
```

Comment:
Isn't it a bit nicer without decalring that class within a class

Reply:
not sure, its purpose is limited to the internals of Uxmapper (hence the name)

Reply:
ok